backend/regalloc/regalloc_gi_utils.ml

```diff
@@ -705,23 +737,29 @@ module Hardware_registers = struct
   let find_available : t -> Reg.t -> Interval.t -> available =
    fun t reg interval ->
     let with_no_overlap =
-      match Lazy.force Selection_heuristics.value with
-      | Selection_heuristics.First_available ->
-        if gi_debug
-        then
-          log ~indent:3
-            "trying to find an available register with 'first-available'";
-        find_first t reg interval
-      | Selection_heuristics.Best_fit ->
-        if gi_debug
-        then
-          log ~indent:3 "trying to find an available register with 'best-fit'";
-        find_using_length t reg interval ~better:( > )
-      | Selection_heuristics.Worst_fit ->
-        if gi_debug
-        then
-          log ~indent:3 "trying to find an available register with 'worst-fit'";
-        find_using_length t reg interval ~better:( < )
+      let rec select heuristic =
```

Comment:
As I understant it, there is at most
to call to `select`, so I think it is a
bit overkill to go through a function
rather than a couple of lets.

Reply:
Sure, changed